src/runtime/typemanager.cs

```diff
@@ -713,6 +713,13 @@ internal static void InitializeSlots(IntPtr type, Type impl)
                         continue;
                     }
 
+                    if (name == "tp_iter" && clrType != null &&
+                        !clrType.GetInterfaces().Any(i => i == typeof(IEnumerable) || i == typeof(IEnumerator)))
+                    {
+                        continue;
+                    }
+
+
```

Comment:
This logic should be in the caller (I guess `CreateType` in this case), and `tp_iter` should be either be made private to not be handled by generic code here, or the slot should be overwritten after `InitializeSlots` is called.

A few reasons:
1) different slots might require different information to figure out if and how they need to be initialized. If in the future we would need to add custom initialization for a different method, the parameter list of this function would grow.
2) if each slot will require own handling, this method would grow too.

Reply:
Yes, my first approach was too hackish.

I went with the second approach of overwriting the slot after `InitializrSlots`. For some reason, I couldn't get moving the handling of tp_iter out of the generic code to work.

Reply:
@danabr its because it is already hackish, as it simply matches public methods to slots by name. I hope we will get rid of it...